src/test-utilities/react-testing.tsx

```diff
@@ -52,53 +49,62 @@ type ReturnedContext = ComplexProviders &
   };
 type Options = DeepPartial<ComplexProviders> & Partial<SimpleProviders>;
 type Context = ReturnedContext;
-interface Props extends ReturnedContext {
+interface Props extends Partial<ReturnedContext> {
   children: React.ReactElement<any>;
+  strict: boolean;
 }
 
 function noop() {}
 
 export function TestProvider({
+  strict,
   children,
-  polaris,
-  themeProvider,
-  frame,
-  intl,
-  scrollLockManager,
-  stickyManager,
-  appBridge,
-  link,
+  themeProvider = createThemeContext(),
+  frame = {
+    showToast: noop,
+    hideToast: noop,
+    setContextualSaveBar: noop,
+    removeContextualSaveBar: noop,
+    startLoading: noop,
+    stopLoading: noop,
+  },
+  intl = new I18n(translations),
+  scrollLockManager = new ScrollLockManager(),
+  stickyManager = new StickyManager(),
+  appBridge = null,
+  link = new Link(undefined),
   ...props
 }: Props) {
   const childWithProps =
     Object.keys(props).length > 0
       ? React.cloneElement(children, props)
       : children;
 
+  const Wrapper = strict? React.StrictMode : React.Fragment;
```

Comment:
Going to have an optional strict mode so this utility can be used for enzyme tests as well